jetbrains-core/src/software/aws/toolkits/jetbrains/core/ProjectLevelCoroutineScopeTracker.kt

```diff
@@ -0,0 +1,29 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.project.Project
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import kotlin.reflect.KClass
+
+class ProjectLevelCoroutineScopeTracker(
+    @SuppressWarnings("UnusedProperty")
+    private val project: Project
+) : Disposable {
+    private val scopes: MutableMap<String, ApplicationThreadPoolScope> = mutableMapOf()
```

Comment:
Should this be a weak hash map?

Are the benefits to caching the scope?

Reply:
changed to weak value map. re: caching, for our use case i think we can go either way